File: app/src/main/java/com/greenaddress/jade/entities/VersionInfo.java
package com.greenaddress.jade.entities;

import com.fasterxml.jackson.annotation.JsonGetter;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonSetter;

@JsonIgnoreProperties(ignoreUnknown=true)
public class VersionInfo {
    private String jadeVersion;
    private Integer jadeOtaMaxChunk;
    private String jadeConfig;
    private String idfVersion;
    private String chipFeatures;
    private String efusemac;
    private Integer jadeFreeHeap;
    private Boolean jadeHasPin;

    @JsonGetter("JADE_VERSION")
    public String getJadeVersion() {
        return jadeVersion;
    }

    @JsonSetter("JADE_VERSION")
    public void setJadeVersion(String jadeVersion) {
        this.jadeVersion = jadeVersion;
    }

    @JsonGetter("JADE_OTA_MAX_CHUNK")
    public Integer getJadeOtaMaxChunk() {
        return jadeOtaMaxChunk;
    }

    @JsonSetter("JADE_OTA_MAX_CHUNK")
    public void setJadeOtaMaxChunk(Integer jadeOtaMaxChunk) {
        this.jadeOtaMaxChunk = jadeOtaMaxChunk;
    }

    @JsonGetter("JADE_CONFIG")
    public String getJadeConfig() {
        return jadeConfig;
    }

    @JsonSetter("JADE_CONFIG")
    public void setJadeConfig(String jadeConfig) {
        this.jadeConfig = jadeConfig;
    }

    @JsonGetter("IDF_VERSION")
    public String getIdfVersion() {
        return idfVersion;
    }

    @JsonSetter("IDF_VERSION")
    public void setIdfVersion(String idfVersion) {
        this.idfVersion = idfVersion;
    }

    @JsonGetter("CHIP_FEATURES")
    public String getChipFeatures() {
        return chipFeatures;
    }

    @JsonSetter("CHIP_FEATURES")
    public void setChipFeatures(String chipFeatures) {
        this.chipFeatures = chipFeatures;
    }

    @JsonGetter("EFUSEMAC")
    public String getEfusemac() {
        return efusemac;
    }

    @JsonSetter("EFUSEMAC")
    public void setEfusemac(String efusemac) {
        this.efusemac = efusemac;
    }

    @JsonGetter("JADE_FREE_HEAP")
    public Integer getJadeFreeHeap() {
        return jadeFreeHeap;
    }

    @JsonSetter("JADE_FREE_HEAP")
    public void setJadeFreeHeap(Integer jadeFreeHeap) {
        this.jadeFreeHeap = jadeFreeHeap;
    }

    @JsonGetter("JADE_HAS_PIN")
    public Boolean getHasPin() {
        return jadeHasPin;
    }

    @JsonSetter("JADE_HAS_PIN")
    public void setJadeHasPin(Boolean jadeHasPin) {
        this.jadeHasPin = jadeHasPin;
    }
}

File: jade/src/main/java/com/blockstream/jade/data/VersionInfo.kt
package com.blockstream.jade.data

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable


@Serializable
data class VersionInfo constructor(
    @SerialName("JADE_VERSION")
    val jadeVersion: String,
    @SerialName("JADE_OTA_MAX_CHUNK")
    val jadeOtaMaxChunk: Int,
    @SerialName("JADE_CONFIG")
    val jadeConfig: String,
    @SerialName("BOARD_TYPE")
    val boardType: String,
    @SerialName("JADE_FEATURES")
    val jadeFeatures: String,
    @SerialName("IDF_VERSION")
    val idfVersion: String,
    @SerialName("CHIP_FEATURES")
    val chipFeatures: String,
    @SerialName("EFUSEMAC")
    val efuseMac: String? = null,
    @SerialName("JADE_STATE")
    val jadeState: JadeState,
    @SerialName("JADE_NETWORKS")
    val jadeNetworks: JadeNetworks,
    @SerialName("JADE_HAS_PIN")
    val jadeHasPin: Boolean,
) : JadeJson<VersionInfo>()